packages/devops/scripts/deployment-common/startBackEnds.sh

```diff
@@ -1,39 +1,60 @@
-#!/bin/bash -le
+#!/usr/bin/env bash
+set -e
 
-DIR=$(dirname "$0")
-TUPAIA_DIR=$DIR/../../../..
-PACKAGES=$(${TUPAIA_DIR}/scripts/bash/getDeployablePackages.sh)
+script_dir=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+tupaia_dir=$(realpath -- "$script_dir"/../../../..)
 
 # Initialise NVM (which sets the path for access to npm, yarn etc. as well)
-. $HOME/.nvm/nvm.sh
+. "$HOME"/.nvm/nvm.sh
 
-# Start back end server packages
-for PACKAGE in ${PACKAGES[@]}; do
-    if [[ $PACKAGE == *server ]]; then
-        if [[ $PACKAGE == 'central-server' ]]; then
-            # reset cwd back to `/tupaia`
-            cd ${TUPAIA_DIR}
-
-            # ensure that the analytics table is fully built
-            echo "Building analytics table"
-            yarn workspace @tupaia/data-api install-mv-refresh
-            yarn workspace @tupaia/data-api patch-mv-refresh up
-            yarn workspace @tupaia/data-api build-analytics-table
-
-            # ensure that the latest permissions based meditrak sync queue has been built
-            yarn workspace @tupaia/central-server create-meditrak-sync-view
+get_backend_packages() {
+    readarray -t deployable_packages < <("$tupaia_dir"/scripts/bash/getDeployablePackages.sh)
+    local filtered=()
+    for package in "${deployable_packages[@]}"; do
+        if [[ $package = *-server ]]; then
+            filtered+=("$package")
         fi
+    done
+    printf '%s\n' "${filtered[@]}"
+}
 
-        # It's a server, start the pm2 process
-        echo "Starting ${PACKAGE}"
-        cd ${TUPAIA_DIR}/packages/$PACKAGE
-        REPLICATION_PM2_CONFIG=''
-        if [ $PACKAGE == "web-config-server" ] || [ $PACKAGE == "report-server" ] ; then
-            # as many replicas as cpu cores - 1
-            REPLICATION_PM2_CONFIG='-i -1'
-        fi
-        pm2 start --name $PACKAGE dist --wait-ready --listen-timeout 15000 --time $REPLICATION_PM2_CONFIG
+readarray -t backend_packages < <(get_backend_packages)
+
+# Start back end server packages
+for package in "${backend_packages[@]}"; do
+    if [[ $package = central-server ]]; then
+        # ensure that the analytics table is fully built
+        echo "Building analytics table"
+        yarn workspace @tupaia/data-api install-mv-refresh
+        yarn workspace @tupaia/data-api patch-mv-refresh up
+        yarn workspace @tupaia/data-api build-analytics-table
+
+        # ensure that the latest permissions based meditrak sync queue has been built
+        yarn workspace @tupaia/central-server create-meditrak-sync-view
     fi
+
+    instances_flag=()
+    if [[ $package = web-config-server || $package = report-server ]]; then
+        # as many replicas as cpu cores - 1
+        instances_flag=(--instances -1)
+    fi
+
+    node_args_flag=''
+    if [[ $package = central-server || $package = tupaia-web-server ]]; then
+        node_args_flag="--node-args='--max-http-header-size=32768'"
```

Comment:
Just a slightly better alternative in my opinion (and it can be consistent with `instances_flag`

```suggestion
        node_args=(--node-args '--max-http-header-size=32768')
```

Then in the pm2 command:
`"${node_args_flag[@]}" \`

Reply:
Perhaps also good to leave a comment why we need to icnrease the max http header here?

Reply:
Yeah, I would actually prefer that too. I actually just followed the PM2 docs, but will be surprised if their CLI can’t parse your version.

Will test this and hopefully go ahead with it 🙂